### DISCUSSION TITLE
Enabling frame pointers across all toolchains

### DISCUSSION THREAD

**Discussion by mwcampbell** [wolfi-dev/os#9984] [Ideas]
Following the lead of Fedora and now Ubuntu, I think it would be a good idea for Wolfi to enable frame pointers across all toolchains (for example, using `-fno-omit-frame-pointer` by default in GCC and clang). I don't think Wolfi's target audience would be particularly sensitive to the marginal performance hit that comes with this change, and it would enable sampling profilers, as well as other benefits for debugging and observability. More background, in the context of the recent Ubuntu announcement: <https://www.polarsignals.com/blog/posts/2023/12/13/embracing-frame-pointers-in-ubuntu-24-04-lts>

**Comment by xnox** [comment]
yes we probably should.

**Comment by kaniini** [comment]
The main area where omitting frame pointers is useful is on register starved architectures (e.g. x86-32).  Wolfi only supports x86-64 and aarch64 which are not register starved.

Wolfi should definitely just add frame pointers.

**Comment by xnox** [comment]
This has now been enabled https://github.com/wolfi-dev/os/pull/21992/files however it will take time to trickle to all the binaries. We are not triggering world rebuild for this alone. Thus we will wait for things to rebuild by themselves for any other reasons. After some time, if things have not been rebuilt, we may trigger mass-rebuild of anything remaining to increase coverage of this feature.
